### ISSUE TITLE
Can't get genericTypeName

### ISSUE THREAD

**Issue by funzin** [krzysztofzablocki/Sourcery#938]
## Overview
- Can't get genericTypeName
- Until `v1.3.1`,  I can get genericTypeName
- From `v1.3.2`, I can't get genericTypeName😢

Attached template file and swift code

## Template&Swift File
e.g. `var` type is `GenericArgumentStruct<Bool>`
```stencil
// can get genericTypeName in v1.3.1
// e.g Bool
{{ var.typeName.generic.typeParameters[0].typeName }} 

// can't get genericTypeName in v1.3.2
{{ var.typeName.generic.typeParameters[0].typeName }}
```

```swift
struct GenericArgumentStruct<T> {
    let value: T
}
```

**Comment by krzysztofzablocki** [comment]
Thanks for the raport, I'll fix it after I finish Sourcery Pro companion app

**Comment by krzysztofzablocki** [comment]
@funzin I'm not seeing the issue on my end, I just tried your code and it works fine, take a look at this template and let me know if I'm doing differently

[Issue_938.sourcerytemplate.zip](https://github.com/krzysztofzablocki/Sourcery/files/6305043/Issue_938.sourcerytemplate.zip)

**Comment by funzin** [comment]
@krzysztofzablocki 

Oh, sorry.
Please check the following another exmaple code.
I can't get typeName.

swift
```swift
struct GenericArgumentStruct<T> {
    let value: T
}

class Foo {
    static let something = GenericArgumentClass<Bool>(value: true)
}

```

stencil
```stencil
{% for variable in type.Foo.variables %}
    {{ variable.name }} has typeParameter {{ variable.typeName.generic.typeParameters[0].typeName }}
{% endfor %}
```

output
```
// Generated using Sourcery 1.3.4 — https://github.com/krzysztofzablocki/Sourcery
// DO NOT EDIT


    something has typeParameter 

```

**Comment by krzysztofzablocki** [comment]
@funzin The issue here is the type inference, that's not provided by Syntax parser and I've manually added support to infer some basic cases, you can workaround by simply adding type attribution to your variable as in my example

**Comment by funzin** [comment]
Thank you for you reply!
will workaround:pray:

**Comment by krzysztofzablocki** [comment]
@funzin you don't need to anymore, it's closed because I fixed already in 1.4 😂

Inference isn't perfect since it's just string manipulation but it should work in most cases, let me know if you run into issues
